server/game_service.py

```diff
@@ -155,9 +157,22 @@ def create_game(
 
     @property
     def live_games(self) -> List[Game]:
-        return [game for game in self.games.values()
+        live_games =  [game for game in self._games.values() 
                 if game.state == GameState.LIVE]
 
+        #FIXME in case the live_games gauge ever gets out of sync with the
```

Comment:
Actually as far as I can tell, this function is never called (and neither is `pending_games`). Although usually I think it would be safer to update the metric with a `set` rather than an `inc/dec` since it won't be able to go out of sync that way.

Reply:
I will rework it such that it only ever calls `set`, at the expense of scanning through the list of active games a lot.
Since it is typically short (<400?) I assume that is acceptably fast.
Counting 1000 items with 25 distinct values and then setting a prometheus gauge to the 25 counts takes 220 microseconds on my laptop. We would be doing that whenever a game is added, removed, or changes its `GameState`.

Reply:
Yea that's the downside. I'm not sure what the right call is. Game lists are short, but games change pretty often. Maybe we could update the stat only in `__init__.py::do_report_dirties`. This triggers once per second which is plenty of resolution.

Reply:
Moving it to `do_report_dirties` sounds great to me!